### ISSUE TITLE
Add selector-max-pseudo-class

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#2805]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

New rule

> Which rule, if any, is this issue related to?

e.g. `selector-max-pseudo-class`

> What CSS is needed to reproduce this issue?

e.g.

```css
span:first-child:hover {
  color: red;
}

.foo > div:nth-child(2):hover {
  color: red;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "selector-max-pseudo-class": 1
  }
}
```

> Which version of stylelint are you using?

e.g. `8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Not required

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Not required

> What did you expect to happen?

Error

> What actually happened (e.g. what warnings or errors you are getting)?

Nothing

```shell
test.css
1:21    Expected 2 spaces (indentation)
```

**Comment by jeddy3** [comment]
@evilebottnawi Are you thinking maximum for the whole selector or for each compound selector? The other `selector-max-*` rules are scoped to the whole selector. So, ideally this rule would be consistent with that, but does that address your use case?

**Comment by evilebottnawi** [comment]
@jeddy3 yep, scoped to the whole selector

**Comment by jeddy3** [comment]
SGTM.

- Name: `selector-max-pseudo-class`
- Primary option: `int`
- Secondary options: none
- Message: "Expected "${selector}" to have no more than ${max} pseudo ${max === 1 ? "class" : "classes"}"
- Section: "Limit language features -> Selector"

**Comment by jeddy3** [comment]
Done in #3195
